### ISSUE TITLE
Question: Binocular mode?

### ISSUE THREAD

**Issue by gmierz** [pupil-labs/pupil#468]
Hi,

I am looking for information about what features binocular mode has? I am mainly wondering if it can already record data from both pupils at the same time. If it does, could you please point me to the code that starts and/or does this? 

Also, is there any documentation about binocular mode anywhere?

Thank you.

**Comment by papr** [comment]
Hey @gmierz !

**TL;DR:** Pupil positions are detected separately and broadcasted over the IPC. The recorder records the results of all plugins including the raw (binocular) pupil data.

Yes, of course, Pupil Capture supports recording both pupils at the same time given, that you are using a binocular headset.

Capture is designed to be able to de/activate the eye cameras one by one. Each eye camera video feed is processed in a separate process. These processes do the pupil detection. The result is send over the [IPC backbone](https://github.com/pupil-labs/pupil/wiki/Pupil-Interprocess-and-Network-Communication) (a message passing system which is used by all our plugins and software components).

The `world` process (which processes the front camera video feed) listens on the IPC and maps newly detected pupil positions to gaze points using a plugin called *gaze mapper*. The binocular gaze mapper also tries to pair affiliated pupil positions. The results are made available to other plugins with an higher `order` (which is a class attribute to manage plugin execution order).

The plugin with the highest `order` is the [recorder](https://github.com/pupil-labs/pupil/blob/master/pupil_src/capture/recorder.py#L82). It therefore gets the raw pupil positions as well as all the other data produced by the preceding plugins. All this data will be stored during a recording. Therefore also binocular pupil data.

So possible files you might want to have a look at are:
- [plugin.py](https://github.com/pupil-labs/pupil/blob/master/pupil_src/shared_modules/plugin.py#L22) (plugin class interface)
- [eye.py](https://github.com/pupil-labs/pupil/blob/master/pupil_src/capture/eye.py) (implementation of the eye process)
- [recorder.py](https://github.com/pupil-labs/pupil/blob/master/pupil_src/capture/recorder.py#L292) (the recorder implementation)
- the [IPC wiki page](https://github.com/pupil-labs/pupil/wiki/Pupil-Interprocess-and-Network-Communication)

**Comment by gmierz** [comment]
Thanks for the quick reply and the great help @papr! This definitely gets me going in the right direction.

**Comment by cpicanco** [comment]
I would rather say TL;RL !!
